app/models/manageiq/providers/cloud_manager/provision/state_machine.rb

```diff
@@ -6,11 +6,37 @@ def create_destination
   def determine_placement
     availability_zone = placement
     options[:dest_availability_zone] = [availability_zone.try(:id), availability_zone.try(:name)]
-    signal :prepare_provision
+    signal :prepare_volumes
+  end
+
+  def prepare_volumes
+    # volumes example: [{name: "vol_name", size: 1}]
+    if miq_request.options[:volumes]
+      phase_context[:requested_volumes] = create_requested_volumes(miq_request.options[:volumes])
+      signal :poll_volumes_complete
+    else
+      signal :prepare_provision
+    end
+  end
+
+  def poll_volumes_complete
+    status, status_message = do_volume_creation_check(phase_context[:requested_volumes])
+    status_message = "completed prepare provision work queued" if status
+    message = "Volume creation is #{status_message}"
+    _log.info(message)
+    update_and_notify_parent(:message => message)
+    if status
+      signal :prepare_provision
+    else
+      requeue_phase
+    end
   end
 
   def start_clone_task
     update_and_notify_parent(:message => "Starting Clone of #{clone_direction}")
+    if phase_context[:requested_volumes]
+      phase_context[:clone_options][:block_device_mapping_v2] = phase_context[:requested_volumes]
```

Comment:
Indeed, so the question is, if in the volume case, we should just hardcode the image there, so it acts exactly the same as image deployment. As a first version, that is fine with me.

Later we could extend it so you could e.g. deploy the image into volume, instead of the local disk and other parameters.